subprojects/docs/src/snippets/userguide/dependencyManagement/customizingResolution/metadataRule/groovy/build.gradle

```diff
@@ -246,6 +246,16 @@ class IvyComponentRule implements ComponentMetadataRule {
 }
 // end::ivy-component-metadata-rule[]
 
+// tag::maven-packaging-component-metadata-rule[]
+class MavenComponentRule implements ComponentMetadataRule {
```

Comment:
Since this rule applies to POM metadata, we can apply it and it will actually run when this snippet is tested, as there are dependencies to libraries with POM metadata in the build script (e.g. Guava). This way we make sure the syntax is correct.

E.g.:

```
dependencies {
    components {
        all(MavenComponentRule)
    }
} 
```